### PR TITLE
iOS build fixes

### LINKED ISSUE #18381: Can't build ios without some modules
##### System information (version)
- OpenCV => 4.4.x (master)
- Operating System / Platform => MAC Catalina
- Compiler => Cmake 3.8
- Python => 2.7
- Xcode => 11.5

##### Detailed description

I'm using [platforms/ios/build_framework.py](https://github.com/opencv/opencv/blob/master/platforms/ios/build_framework.py) to build opencv for ios swift project.
When I run the script(build_framework.py) without any arguments, it works well.
But if I put arguments **--without**, the build fails.

I checked this issue in Xcode 11.5, 11.7, 12.0 and all the same result.
The error says
```
CMake Error at modules/objc/generator/CMakeLists.txt:86 (add_dependencies):
Cannot add target-level dependencies to non-existent target "opencv_world".
```

This [line](https://github.com/opencv/opencv/blob/f7b8f522ff4084d584628811a02a621736c3a731/platforms/ios/build_framework.py#L186) of code makes change if I use argument **--without**.

##### Steps to reproduce

```bash
python platforms/ios/build_framework.py build_ios_without_doublecheck_2 --without calib3d
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

### PR DESCRIPTION
**Pull request by komakai**
This PR fixes #18381 and #18405
 - fix cmake file to handle case when we are not building **opencv_world**
 - modify **install_name** to not hard code library path
 - update tests and samples to work with the change to **install_name**

<cut/>

### Pull Request Readiness Checklist

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.

```
allow_multiple_commits=1
```

### CODE REVIEW

File: modules/objc/generator/CMakeLists.txt [hunk 1]
```diff
@@ -83,4 +83,10 @@ add_custom_target(gen_opencv_objc_source DEPENDS ${objc_generated_files}
             "${CMAKE_CURRENT_BINARY_DIR}/gen_objc.json"
 )
 
-add_dependencies(opencv_world gen_opencv_objc_source)
+if(BUILD_opencv_world)
+  add_dependencies(opencv_world gen_opencv_objc_source)
+else()
+  foreach(m ${OPENCV_OBJC_MODULES})
+    add_dependencies(${m} gen_opencv_objc_source)
```
Comment (alalek) [inline modules/objc/generator/CMakeLists.txt:90]: Are other modules depend on generated code?

Did you try?
```
add_dependencies(${the_module} gen_opencv_objc_source)
```

**Review by alalek — COMMENTED** [review]
Thank you for contribution!

File: modules/objc/generator/CMakeLists.txt (hunk 1 above)
Comment (komakai) [reply modules/objc/generator/CMakeLists.txt]: OK - I've figured it out now. All I needed to do was add the target **gen_opencv_objc_source** to the ALL target

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**